src/test/java/io/logz/LogzioLogbackAppenderTest.java

```diff
@@ -0,0 +1,320 @@
+        Marker marker = MarkerFactory.getMarker(markerTestValue);
+
+        Logger testLogger = createLogger(token, type, loggerName, drainTimeout, false, null);
+
+        testLogger.info(marker, message1);
+
+        sleepSeconds(2 * drainTimeout);
+
+        mockListener.assertNumberOfReceivedMsgs(1);
+        MockLogzioBulkListener.LogRequest logRequest = mockListener.assertLogReceivedByMessage(message1);
+        mockListener.assertLogReceivedIs(logRequest, token, type, loggerName, Level.INFO.levelStr);
+
+        assertThat(logRequest.getStringFieldOrNull(markerKey)).isEqualTo(markerTestValue);
+    }
+
+    @Test
+    public void testContextReset() throws Exception {
+        logger.info("context.reset() is called when logback loads a new logback.xml in-flight");
+        String token = "testingContextReset";
+        String type = "contextResetType";
+        String loggerName = "ContextResetLogger";
+        int drainTimeout = 1;
+
+        String message1 = "Before Reset Line - "+random(5);
+        String message2 = "After Reset Line - "+random(5);
+
+        Logger testLogger = createLogger(token, type, loggerName, drainTimeout, false, null);
+
+        testLogger.info(message1);
+        sleepSeconds(2 * drainTimeout);
+
+        mockListener.assertNumberOfReceivedMsgs(1);
+        MockLogzioBulkListener.LogRequest logRequest = mockListener.assertLogReceivedByMessage(message1);
+        mockListener.assertLogReceivedIs(logRequest, token, type, loggerName, Level.INFO.levelStr);
+
+        LoggerContext loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
+
+        // This clears everything from the context
+        loggerContext.reset();
+
+        // We need to add the appender again
+        testLogger = createLogger(token, type, loggerName, drainTimeout, false, null);
+
+        testLogger.warn(message2);
+        sleepSeconds(2 * drainTimeout);
+
+        mockListener.assertNumberOfReceivedMsgs(2);
+        logRequest = mockListener.assertLogReceivedByMessage(message2);
+        mockListener.assertLogReceivedIs(logRequest, token, type, loggerName, Level.WARN.levelStr);
+    }
+
+    @Test
+    public void testTokenAndLogzioUrlFromSystemEnvironment() {
+        String token = System.getenv("JAVA_HOME");
+        String type = "testType";
+        String loggerName = "testLogger";
+        int drainTimeout = 1;
+
+        String message1 = "Just a log - " + random(5);
+        Logger testLogger = createLogger("$JAVA_HOME", type, loggerName, drainTimeout, false, null);
+
+        testLogger.info(message1);
+
+        sleepSeconds(2 * drainTimeout);
+
+        mockListener.assertNumberOfReceivedMsgs(1);
+        MockLogzioBulkListener.LogRequest logRequest = mockListener.assertLogReceivedByMessage(message1);
+        mockListener.assertLogReceivedIs(logRequest, token, type, loggerName, Level.INFO.levelStr);
+    }
+
+    protected Logger createLogger(String token, String type, String loggerName, Integer drainTimeout,
```

Comment:
This is copy pasted from the test class before - How about creating a base class for the tests containing both the start / stop of the mock listener and the shared logic such as createLogger and sleepSeconds()?